### PR TITLE
glib: Various paramspec builder API improvements

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

File: glib/src/gobject/binding_group.rs [hunk 1]
```diff
@@ -426,20 +426,13 @@ mod test {
...
-                            None,
-                            false,
-                            crate::ParamFlags::READWRITE | crate::ParamFlags::EXPLICIT_NOTIFY,
-                        ),
+                        crate::ParamSpecString::builder("name")
+                            .explicit_notify()
+                            .build(),
+                        crate::ParamSpecBoolean::builder("enabled")
+                            .explicit_notify()
+                            .default_value(false)
```
[the remark this answers is not in the document]
Comment (sdroege) [reply glib/src/gobject/binding_group.rs]: Yes

File: glib/src/param_spec.rs [hunk 2]
```diff
@@ -1315,16 +1646,67 @@ impl ParamSpecVariant {
...
+    fn set_blurb(&mut self, blurb: Option<&'a str>) {
+        self.blurb = blurb;
+    }
+    fn set_flags(&mut self, flags: crate::ParamFlags) {
+        self.flags = flags;
+    }
+    fn current_flags(&self) -> crate::ParamFlags {
+        self.flags
+    }
+}
```
Comment (bilelmoussaoui) [inline glib/src/param_spec.rs:1709]: completely random but maybe we can make this trait implemented by a small macro if you want ? :)

File: glib/src/subclass/object.rs [hunk 3]
```diff
@@ -279,34 +280,15 @@ mod test {
...
-                            Some("Child object"),
-                            super::ChildObject::static_type(),
-                            crate::ParamFlags::READWRITE,
-                        ),
+                        crate::ParamSpecString::builder("name").build(),
+                        crate::ParamSpecString::builder("construct-name")
+                            .construct_only()
+                            .build(),
+                        crate::ParamSpecBoolean::builder("constructed")
+                            .default_value(false)
```
Comment (bilelmoussaoui) [inline glib/src/subclass/object.rs:288]:
```suggestion
```

File: glib/src/param_spec.rs (hunk 2 above)
Comment (sdroege) [reply glib/src/param_spec.rs:1709]: Not so easy because some builders are `<'a>`, others are `<'a, T>` etc

Comment (sdroege) [reply glib/src/param_spec.rs:1709]: I'd keep it like this for now.

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**